src/widget.js

```diff
@@ -97,6 +74,45 @@ export default class Widget extends Plugin {
 		}, { priority: 'high' } );
 	}
 
+	/**
+	 * Model to view selection converter.
+	 * Converts selection placed over a widget element to the fake selection.
+	 *
+	 * @private
+	 * @param {module:utils/eventinfo~EventInfo} eventInfo
+	 * @param {Object} data Additional information about the change.
+	 * @param {Object} conversionApi Conversion interface.
+	 */
+	_dispatchDowncastSelection( eventInfo, data, conversionApi ) {
+		// Remove selected class from previously selected widgets.
+		this._clearPreviouslySelectedWidgets( conversionApi.writer );
+
+		const viewWriter = conversionApi.writer;
+		const viewSelection = viewWriter.document.selection;
+		const selectedElement = viewSelection.getSelectedElement();
+		let lastMarked;
+
+		for ( const range of viewSelection.getRanges() ) {
+			for ( const value of range ) {
+				const node = value.item;
+
+				if ( node.is( 'element' ) && isWidget( node ) ) {
```

Comment:
As I [commented](https://github.com/ckeditor/ckeditor5-widget/issues/57#issuecomment-436394820), this code was just an ugly quick fix. I didn't mean the fact that it was defined in that listener – moving this code to a method doesn't change much (in fact, I'd revert this change – less code, less documentation). What I meant by "it needs refactoring" is that I added a nested `if()` for no good reason – you can flatten this and extract the condition to a helper function. Also, I think we could use the last element from `previouslySelected` if it was an array (although, I have mixed feeling about such change). Plus, the added condition requires a comment because `Array.from( node.getAncestors() ).includes( lastMarked )` is hardly idiomatic.